code/datums/bug_report.dm

```diff
@@ -0,0 +1,161 @@
+	GLOB.bug_reports -= src
+	return ..()
+
+// returns the body payload
+/datum/tgui_bug_report_form/proc/create_form()
+	var/list/testmerges = world.TgsTestMerges()
+	var/text_output_tm = ""
+	for(var/tm in testmerges)
+		text_output_tm += "[tm]"
+
+	var/desc = {"
+### Testmerges
+[text_output_tm]
+
+### Round ID
+[GLOB.round_id]
+
+### Description of the bug
+[bug_report_data["description"]]
+
+### What's the difference with what should have happened?
+[bug_report_data["expected_behavior"]]
+
+### How do we reproduce this bug?
+[bug_report_data["steps"]]
+	"}
+
+	return desc
+
+// the real deal, we are sending the request through the api.
+/datum/tgui_bug_report_form/proc/send_request(payload_body, client/user)
+	// for any future changes see https://docs.github.com/en/rest/issues/issues
+	var/repo_name = CONFIG_GET(string/repo_name)
+	var/org = CONFIG_GET(string/org)
+	var/token = CONFIG_GET(string/github_app_api)
+
+	if(!token || !org || !repo_name)
+		tgui_alert(user, "The configuration is not set for the external api", "Issue not reported!")
+		external_link_prompt(user)
+		qdel(src)
+		return
+
+	var/url = "https://api.github.com/repos/[org]/[repo_name]/issues"
+	var/list/headers = list()
+
+	headers["Authorization"] = "Bearer [token]"
+	headers["Content-Type"] = "text/markdown; charset=utf-8"
+	headers["Accept"] = "application/vnd.github+json"
+
+	var/datum/http_request/request = new
+
+	var/list/payload = list(
+		"title" = bug_report_data["title"],
+		"body" = payload_body,
+	)
+
+	request.prepare(RUSTG_HTTP_METHOD_POST, url, json_encode(payload), headers)
+	request.begin_async()
+	UNTIL(request.is_complete())
+	var/datum/http_response/response = request.into_response()
+
+	if(response.errored || response.status_code != STATUS_SUCCESS)
+		external_link_prompt(user)
+	else
+		message_admins("[user.ckey] has approved a bug report from [initial_user.ckey] titled [bug_report_data["title"]] at [time2text(world.timeofday, "YYYY-MM-DD hh:mm:ss")].")
+		to_chat(initial_user, SPAN_WARNING("An admin has successfully submitted your report and it should now be visible on GitHub, Thanks again!"))
+	qdel(src)// approved and submitted, we no longer need the datum.
+
+// proc that creates a ticket for an admin to approve or deny a bug report request
+/datum/tgui_bug_report_form/proc/bug_report_request()
+	to_chat(initial_user, SPAN_WARNING("Your bug report has been submitted, thank you!"))
+
+	GLOB.bug_reports += src
+	var/general_message = "[initial_user.ckey] has created a bug report, you may modify the report to your liking before submitting it to Github"
+	GLOB.admin_help_ui_handler.perform_adminhelp(initial_user, general_message, urgent = FALSE)
+	var/href_message = ADMIN_VIEW_BUG_REPORT(src)
+	initial_user.current_ticket.AddInteraction(href_message)
+
+/datum/tgui_bug_report_form/ui_act(action, list/params, datum/tgui/ui)
+	. = ..()
+	if (.)
+		return
+	var/mob/user = ui.user
+	switch(action)
+		if("confirm")
+			bug_report_data = params
```

Comment:
can we do some additional sanity checks on this? this is later sent raw to admins, so could be sending intentionally malformed XSS text to admins chats, which is bad

Reply:
done